client/result/result.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```

Comment:
Agreed that having `client` depend on `bootstrap` could lead to circular dependency issues if not careful, so I'm not opposed to moving `result` to a separate package that both `client` and `bootstrap` depend on. My main argument is to not have that live under `client/result`, since in most cases a nested package either is closed tied to or implements interfaces in its parent package, whereas in this case `result` is neither to the `client` package, so have the options and types related to bootstrapping results live in `client/result` feels wrong to me. Instead, `result` is logically an integral part of the bootstrap process, so IMO a better option would be moving it to `bootstrap/result`.